src/main/java/core/basesyntax/FigureSupplierImplementation.java

```diff
@@ -0,0 +1,42 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplierImplementation implements FigureSupplier {
+    private Random random = new Random();
+
+    @Override
+    public Figure getRandomFigure() {
+        var shape = Shapes.values()[random.nextInt(Shapes.values().length)];
+        switch (shape) {
+            case RECTANGLE -> {
+                return new Rectangle(random.nextDouble(), random.nextDouble());
```

Comment:
Creating a figure, don't pass expressions in the constructor. Create separate variables and pass them on for better code readability.